lib/api.js

```diff
@@ -314,6 +315,10 @@ function transformOptions(options, encoding) {
     transformed.windowOptions.pretendToBeVisual = Boolean(options.pretendToBeVisual);
   }
 
+  if (options.ResourceLoader) {
+    transformed.windowOptions.ResourceLoader = options.ResourceLoader;
+  }
```

Comment:
> The option should be lowercase resourceLoader, and accept an instance of ResourceLoader, instead of accepting a constructor. This is similar to the virtualConsole or cookieJar options.

Ok. I wasn't what was the best, if the Constructor or the instance.

> Ideally a resource loader should be able to handle multiple documents/windows. This may involve creating a new internal class, something like PerDocumentResourceLoader, which gets spawned from the ResourceLoader (using a "private" method). But users of jsdom should not worry about this, if possible.

Sorry, I don't understand this 100% hehehe.

Do you mean something like this?
`class PerDocumentResourceLoader {
   constructor(resourceLoader, document) {
     this.document = document;
     this.resourceLoader = resourceLoader;
   }
   
   fetch(url, options) {
     // do some job with options
     return this.resourceLoader(url, newOptions);
   }  
}`

Reply:
Something like that, yeah. The idea being that fetch() on PDRL can use Document-specific stuff like tracking Window shutdown.

Reply:
An alternate design, maybe simpler for now, is to just add document to fetch()s options. Each call site then needs to pass it in, but that isn't too bad.

Reply:
I will try the `PerDocumentResourceLoader` if I see that things get complicated, I can go back and use pass document as an option.

But I prefer the PerDocumentResourceLoader, because if someone create a custom resource loader and they don't use `super.fetch` then they will lose the Document-specific stuff